RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc

```diff
@@ -107,39 +108,33 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
       auto const gind = genericDet->index();
       for (auto const& px : det) {
         digis_h.view()[nDigis].moduleId() = uint16_t(gind);
-
         digis_h.view()[nDigis].xx() = uint16_t(px.row());
         digis_h.view()[nDigis].yy() = uint16_t(px.column());
         digis_h.view()[nDigis].adc() = uint16_t(px.adc());
-
         digis_h.view()[nDigis].clus() = 0;
-
         digis_h.view()[nDigis].pdigi() = uint32_t(px.packedData());
-
         digis_h.view()[nDigis].rawIdArr() = uint32_t(detid);
-
-        nDigis++;
+        ++nDigis;
       }
     }
 
-    digis_d = SiPixelDigisSoACollection(nDigis, iEvent.queue());
-    alpaka::memcpy(iEvent.queue(), digis_d.buffer(), digis_h.buffer());
-
-    Algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d.view(), nDigis_);
+    digis_d_ = SiPixelDigisSoACollection(nDigis, iEvent.queue());
+    alpaka::memcpy(iEvent.queue(), digis_d_->buffer(), digis_h.buffer());
+    Algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d_->view(), nDigis_);
   }
 
   void SiPixelPhase2DigiToCluster::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
     if (nDigis_ == 0) {
       SiPixelClustersSoACollection clusters_d{pixelTopology::Phase2::numberOfModules, iEvent.queue()};
-      SiPixelDigisSoACollection digis_d_zero{nDigis_, iEvent.queue()};
+      SiPixelDigisSoACollection digis_d_zero{0, iEvent.queue()};
```

Comment:
@AdrianoDee do you know why here we use a different collection instead of using `digis_d_` ?